packages/@n8n/db/src/repositories/folder.repository.ts

```diff
@@ -102,7 +129,7 @@ export class FolderRepository extends Repository<FolderWithWorkflowAndSubFolderC
 		}
 
 		if (select?.workflowCount) {
-			query.loadRelationCountAndMap('folder.workflowCount', 'folder.workflows');
```

Comment:
Wondering if something like this would simplify things. Adding the filter directly in the `loadRelationCountAndMap`

```
    .loadRelationCountAndMap(
        'user.itineraryCount',
        'user.post',
        'post',
        qb => qb.where('post.type =:type', { type: 'itinerary' }),
    )
```

Reply:
`applyWorkflowCountSelect` is doing that

```
query.loadRelationCountAndMap('folder.workflowCount', 'folder.workflows', 'workflow', (qb) =>
				qb.andWhere('workflow.isArchived = :isArchived', {
					isArchived: isArchivedFilter,
				}),
			);
```

but the extra filter can only be applied if the `isArchivedFilter` is boolean true/false, if it is undefined the where condition has to be skipped for the logic to work and that's what the `applyWorkflowCountSelect` wraps 🤔